src/main/java/org/entando/kubernetes/validator/PluginDescriptorValidator.java

```diff
@@ -298,24 +338,47 @@ private boolean isSecretKeyRefValid(SecretKeyRef secretKeyRef) {
 
         return !ObjectUtils.isEmpty(secretKeyRef.getName())
                 && secretKeyRef.getName().length() <= 253
-                && ! ObjectUtils.isEmpty(secretKeyRef.getKey())
+                && !ObjectUtils.isEmpty(secretKeyRef.getKey())
                 && DNS_LABEL_HOST_REGEX_PATTERN.matcher(secretKeyRef.getName()).matches();
     }
 
 
+    /**
+     * validate the fullDeploymentName. if the validation fails an EntandoComponentManagerException is thrown
+     *
+     * @param descriptor the descriptor of which validate the fullDeploymentName length
+     * @return the validated string
+     */
+    private PluginDescriptor validateFullDeploymentNameLength(PluginDescriptor descriptor) {
+        if (descriptor.getFullDeploymentName().length() > fullDeploymentNameMaxlength) {
+
+            throw new EntandoComponentManagerException(
+                    String.format(
+                            DEPLOYMENT_BASE_NAME_MAX_LENGHT_EXCEEDED_ERROR,
+                            descriptor.getFullDeploymentName(),
+                            fullDeploymentNameMaxlength));
+        }
+
+        return descriptor;
+    }
+
+
     public static final String SECURITY_LEVEL_NOT_RECOGNIZED =
             "The received plugin descriptor contains an unknown securityLevel. Accepted values are: "
                     + Arrays.stream(PluginSecurityLevel.values()).map(PluginSecurityLevel::toName)
                     .collect(Collectors.joining(", "));
     public static final String VERSION_NOT_VALID =
             "The plugin %s descriptor contains an invalid descriptorVersion. Accepted versions are: "
                     + Arrays.stream(PluginDescriptorVersion.values()).map(PluginDescriptorVersion::getVersion)
-                            .collect(Collectors.joining(", "));
+                    .collect(Collectors.joining(", "));
     public static final String ENV_VARS_NOT_VALID =
             "The descriptor of the %s plugin contains an invalid environment variable (number %d). Rules are:\n"
                     + "1) each environment variable must have a name\n"
                     + "2) each environment variable must have a value OR a SecretKeyRef fully populated\n"
                     + "3) the name of a Secret object must be a valid DNS subdomain name";
+    public static final String NON_OWNED_SECRET =
+            "The descriptor of the %s plugin contains an invalid environment variable (number %d) that points to a "
+                    + "secret non owned by the plugin. Valid secret names end with the bundle id %s";
```

Comment:
missing space

Reply:
@cecchisandrone do you mean the one before the word secret that is at the end of the previous line or another one that I'm still missing?

Reply:
nevermind my mistake